Build failure: python3Packages.clldutils

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

```
============================= test session starts ==============================
platform linux -- Python 3.13.4, pytest-8.3.5, pluggy-1.6.0
rootdir: /build/source
configfile: setup.cfg
testpaths: tests
plugins: cov-nixpkgs-stub-1.0.0, mock-3.14.1
collected 224 items / 1 error                                                  

==================================== ERRORS ====================================
___________________ ERROR collecting tests/test_declenum.py ____________________
tests/test_declenum.py:3: in <module>
    from clldutils.declenum import DeclEnum
/nix/store/w67iljbhj5f5v6nd6n6s588wjmn0ksjv-python3.13-clldutils-3.21.0/lib/python3.13/site-packages/clldutils/declenum.py:59: in <module>
    class DeclEnum(metaclass=EnumMeta):
/nix/store/w67iljbhj5f5v6nd6n6s588wjmn0ksjv-python3.13-clldutils-3.21.0/lib/python3.13/site-packages/clldutils/declenum.py:51: in __init__
    sym = reg[v[0]] = EnumSymbol(cls, k, *v)
E   TypeError: EnumSymbol.__init__() missing 2 required positional arguments: 'value' and 'description'
=========================== short test summary info ============================
ERROR tests/test_declenum.py - TypeError: EnumSymbol.__init__() missing 2 required positional arguments: '...
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.43s ===============================
error: builder for '/nix/store/z4ay9ab3jva1mkzbs22y5m3paqzcqglj-python3.13-clldutils-3.21.0.drv' failed with exit code 2;
```
as of d0aef5b0a2049479dc5fd4a1a0334ee6a2a1e9b9

### Can Hydra reproduce this build failure?

Please select the Hydra Status.

### Link to Hydra build job

_No response_

### Relevant log output

```console
============================= test session starts ==============================
platform linux -- Python 3.13.4, pytest-8.3.5, pluggy-1.6.0
rootdir: /build/source
configfile: setup.cfg
testpaths: tests
plugins: cov-nixpkgs-stub-1.0.0, mock-3.14.1
collected 224 items / 1 error                                                  

==================================== ERRORS ====================================
___________________ ERROR collecting tests/test_declenum.py ____________________
tests/test_declenum.py:3: in <module>
    from clldutils.declenum import DeclEnum
/nix/store/w67iljbhj5f5v6nd6n6s588wjmn0ksjv-python3.13-clldutils-3.21.0/lib/python3.13/site-packages/clldutils/declenum.py:59: in <module>
    class DeclEnum(metaclass=EnumMeta):
/nix/store/w67iljbhj5f5v6nd6n6s588wjmn0ksjv-python3.13-clldutils-3.21.0/lib/python3.13/site-packages/clldutils/declenum.py:51: in __init__
    sym = reg[v[0]] = EnumSymbol(cls, k, *v)
E   TypeError: EnumSymbol.__init__() missing 2 required positional arguments: 'value' and 'description'
=========================== short test summary info ============================
ERROR tests/test_declenum.py - TypeError: EnumSymbol.__init__() missing 2 required positional arguments: '...
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.43s ===============================
error: builder for '/nix/store/z4ay9ab3jva1mkzbs22y5m3paqzcqglj-python3.13-clldutils-3.21.0.drv' failed with exit code 2;
       last 25 log lines:
       > Executing pythonImportsCheckPhase
       > Running phase: pytestCheckPhase
       > Executing pytestCheckPhase
       > pytest flags: -m pytest
       > ============================= test session starts ==============================
       > platform linux -- Python 3.13.4, pytest-8.3.5, pluggy-1.6.0
       > rootdir: /build/source
       > configfile: setup.cfg
       > testpaths: tests
       > plugins: cov-nixpkgs-stub-1.0.0, mock-3.14.1
       > collected 224 items / 1 error                                                  
       >
       > ==================================== ERRORS ====================================
       > ___________________ ERROR collecting tests/test_declenum.py ____________________
       > tests/test_declenum.py:3: in <module>
       >     from clldutils.declenum import DeclEnum
       > /nix/store/w67iljbhj5f5v6nd6n6s588wjmn0ksjv-python3.13-clldutils-3.21.0/lib/python3.13/site-packages/clldutils/declenum.py:59: in <module>
       >     class DeclEnum(metaclass=EnumMeta):
       > /nix/store/w67iljbhj5f5v6nd6n6s588wjmn0ksjv-python3.13-clldutils-3.21.0/lib/python3.13/site-packages/clldutils/declenum.py:51: in __init__
       >     sym = reg[v[0]] = EnumSymbol(cls, k, *v)
       > E   TypeError: EnumSymbol.__init__() missing 2 required positional arguments: 'value' and 'description'
       > =========================== short test summary info ============================
       > ERROR tests/test_declenum.py - TypeError: EnumSymbol.__init__() missing 2 required positional arguments: '...
       > !!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
       > =============================== 1 error in 0.43s ===============================
       For full logs, run:
         nix log /nix/store/z4ay9ab3jva1mkzbs22y5m3paqzcqglj-python3.13-clldutils-3.21.0.drv
error: 1 dependencies of derivation '/nix/store/3sam9ybxg6mp3zw9vi2b5ggh18d4pail-python3.13-segments-2.2.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/srnv8j3p97i8pbi5sr68wzsqq5grnka1-python3.13-phonemizer-3.3.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ydfqf8sq6104727phx7hmk1y58pzwzvx-python3.13-kokoro-0-unstable-2025-06-16.drv' failed to build
error: 1 dependencies of derivation '/nix/store/pxd6lns98rblhj32miy1vlyla8rd61zi-gtk-audio-player-0.1.0.drv' failed to build
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.33, NixOS, 25.05 (Warbler), 25.05.804113.6c64dabd3aa8`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.3`
 - channels(root): `"nixos-25.05"`
 - nixpkgs: `/nix/store/s8xy39429qidvbjx7g2vlipx3a8hz0fh-nixos-25.05/nixos`

### Notify maintainers

@mattmelling 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/bibtexparser/2.nix
+++ b/pkgs/development/python-modules/bibtexparser/2.nix
@@ -1,0 +1,37 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  pylatexenc,
+  pytestCheckHook,
+  setuptools,
+}:
+
+buildPythonPackage rec {
+  pname = "bibtexparser";
+  version = "2.0.0b8";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "sciunto-org";
+    repo = "python-bibtexparser";
+    tag = "v${version}";
+    hash = "sha256-531Mh/5DUYayXm1H0v4dPX0P9mRcqcQcU/A+f4wwqxg=";
+  };
+
+  build-system = [ setuptools ];
+
+  dependencies = [ pylatexenc ];
+
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  pythonImportsCheck = [ "bibtexparser" ];
+
+  meta = {
+    description = "Bibtex parser for Python";
+    homepage = "https://github.com/sciunto-org/python-bibtexparser";
+    changelog = "https://github.com/sciunto-org/python-bibtexparser/blob/${src.tag}/CHANGELOG";
+    license = lib.licenses.mit;
+    maintainers = [ lib.maintainers.amadejkastelic ];
+  };
+}

--- a/pkgs/development/python-modules/clldutils/default.nix
+++ b/pkgs/development/python-modules/clldutils/default.nix
@@ -4,11 +4,11 @@
   buildPythonPackage,
   colorlog,
   fetchFromGitHub,
+  bibtexparser_2,
   git,
   lxml,
   markdown,
   markupsafe,
-  mock,
   postgresql,
   pylatexenc,
   pytest-cov-stub,
@@ -22,7 +22,7 @@
 
 buildPythonPackage rec {
   pname = "clldutils";
-  version = "3.21.0";
+  version = "3.24.2";
   pyproject = true;
   disabled = pythonOlder "3.8";
 
@@ -30,13 +30,14 @@ buildPythonPackage rec {
     owner = "clld";
     repo = "clldutils";
     rev = "v${version}";
-    hash = "sha256-OD+WJ9JuYZb/oXDgVqL4i5YlcVEt0+swq0SB3cutyRo=";
+    hash = "sha256-xIs6Lq9iDdcM3j51F27x408oUldvy5nlvVdbrAS5Jz0=";
   };
 
   nativeBuildInputs = [ setuptools ];
 
   propagatedBuildInputs = [
     attrs
+    bibtexparser_2
     colorlog
     lxml
     markdown
@@ -47,7 +48,6 @@ buildPythonPackage rec {
   ];
 
   nativeCheckInputs = [
-    mock
     postgresql
     pytest-cov-stub
     pytest-mock

--- a/pkgs/development/python-modules/clldutils/default.nix
+++ b/pkgs/development/python-modules/clldutils/default.nix
@@ -24,18 +24,17 @@ buildPythonPackage rec {
   pname = "clldutils";
   version = "3.24.2";
   pyproject = true;
-  disabled = pythonOlder "3.8";
 
   src = fetchFromGitHub {
     owner = "clld";
     repo = "clldutils";
-    rev = "v${version}";
+    tag = "v${version}";
     hash = "sha256-xIs6Lq9iDdcM3j51F27x408oUldvy5nlvVdbrAS5Jz0=";
   };
 
-  nativeBuildInputs = [ setuptools ];
+  build-system = [ setuptools ];
 
-  propagatedBuildInputs = [
+  dependencies = [
     attrs
     bibtexparser_2
     colorlog
@@ -55,11 +54,11 @@ buildPythonPackage rec {
     git
   ];
 
-  meta = with lib; {
-    changelog = "https://github.com/clld/clldutils/blob/${src.rev}/CHANGES.md";
+  meta = {
+    changelog = "https://github.com/clld/clldutils/blob/${src.tag}/CHANGES.md";
     description = "Utilities for clld apps without the overhead of requiring pyramid, rdflib et al";
     homepage = "https://github.com/clld/clldutils";
-    license = licenses.asl20;
-    maintainers = with maintainers; [ melling ];
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ melling ];
   };
 }

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -1817,6 +1817,8 @@ self: super: with self; {
 
   bibtexparser = callPackage ../development/python-modules/bibtexparser { };
 
+  bibtexparser_2 = callPackage ../development/python-modules/bibtexparser/2.nix { };
+
   bidict = callPackage ../development/python-modules/bidict { };
 
   bids-validator = callPackage ../development/python-modules/bids-validator { };

